microsoft-edge/devtools-guide-chromium/experimental-features/index.md

```diff
@@ -580,23 +603,45 @@ See also:
 * [chrome.runtime](https://developer.chrome.com/extensions/runtime) - retrieves the background page.
 
 Status:
-*  This checkbox is present in Microsoft Edge Canary 116.
-*  This checkbox is present in Microsoft Edge Stable 114.
+*  This checkbox is present in Microsoft Edge Canary 117.
+*  This checkbox is present in Microsoft Edge Stable 115.
+
+
+<!-- ====================================================================== -->
+## Show warning about Self-XSS when pasting code
+<!-- no (?) link -->
+
+_new as of Microsoft Edge 117_
+
+<!-- _Self cross-site scripting_ (Self-XSS) is a security attack vector that relies on bad actors tricking unsuspecting users into attacking themselves by running code.  For example, a bad actor could create a website that asks users to copy some JavaScript code into their clipboard, then open the Console tool, paste that code into the Console, and then press **Enter**.  This can allow the bad actor to run code on the user's device with the user's privileges. -->
+
+Displays a warning about self cross-site scripting whenever code is pasted into the Console.  If this checkbox is enabled, when the user pastes code into the Console tool, a warning message appears in the Console, such as the following:
```

Comment:
@captainbrosset @zoherghadyali -- 
* "whenever code is pasted" - does the warning appear every time you paste code into Console, or only 1 time?

Reply:
Ah, I got it to work in Edge Canary. It didn't work for me yesterday.
To test it yourself:

1. Open Edge Canary, DevTools, enable the experiment, and reload DevTools.
2. Open the Console.
3. Copy this code `localStorage.setItem("myCat", "Tom");` (or any other code snippet) and try to paste it into the Console.

Result: the code is **not** pasted, and a warning message appears instead:

> Warning: Do not paste code you do not understand or have not checked yourself into the DevTools console. This could allow attackers to steal your identity or take control of your computer. Please type 'allow pasting' below to allow pasting.

This happens every time. So if you keep on trying to paste the same code snippet, the message will appear again (now, because there's a grouping mechanism in the Console, if you paste several times in a row, a little counter will appear instead, but if you had other messages appear in the meantime, then a new warning would appear).

<img width="721" alt="image" src="https://github.com/MicrosoftDocs/edge-developer/assets/1152698/4112c2cc-dafa-4b36-b681-11fffa96323e">

Reply:
my testing:
after entering "allow pasting", close browser, open browser, paste code into Console -> message no longer appears.